packages/components/button/src/IconButton/IconButton.tsx

```diff
@@ -49,24 +67,44 @@ function _IconButton<
     icon,
     className,
     size = 'medium',
+    withTooltip = false,
+    tooltipProps,
+    'aria-label': ariaLabel,
     ...otherProps
   } = props;
 
   const density = useDensity();
 
   const styles = getStyles({ size, density });
 
-  return (
+  const iconButtton = (
     <Button
       testId={testId}
       ref={ref}
       variant={variant}
       className={cx(styles.iconButton, className)}
       size={size}
-      {...otherProps}
       startIcon={icon}
+      aria-label={ariaLabel}
+      {...otherProps}
     />
   );
+
+  if (withTooltip) {
+    const {
+      showDelay = 600,
+      content = ariaLabel,
+      ...restTooltipProps
```

Comment:
not a biggie or anything, but in other places we usually use other rather than rest